app/src/main/java/io/pslab/activity/WaveGeneratorActivity.java

```diff
@@ -201,6 +221,11 @@ protected void onCreate(Bundle savedInstanceState) {
 
         removeStatusBar();
 
+        realm = LocalDataLog.with().getRealm();
+        gpsLogger = new GPSLogger(this,
```

Comment:
I don't understand why do we need to plot coordinates for saving a wave generator configuration? If oscilloscope has it I could understand but wave generator?

Reply:
I just added this to have consistency and also when we display the log in data logger activity there is an option to display coordinates on the map for all instruments, that why i added it here as well

Reply:
secondly, if we don't have the map data, the map icon should change to another icon like the open folder icon.

Reply:
That can be done. But i Think we can make this changes in a separate PR, since those changes will be applied to all the instruments. I don't see any harm right now to keep coordinates in wave generator